packages/edit-post/src/components/layout/index.js

```diff
@@ -142,6 +147,13 @@ function Layout( { styles } ) {
 			hasBlockSelected ? 'edit-post/block' : 'edit-post/document'
 		);
 
+	// Check if the block list view should be open by default.
+	useEffect( () => {
+		if ( isListViewOpenByDefault ) {
+			setIsListViewOpened( true );
+		}
+	}, [] );
```

Comment:
I think we should start respecting the [exhaustive deps](https://github.com/facebook/react/issues/14920#issuecomment-471070149) lint rule. There's an open PR and discussion in https://github.com/WordPress/gutenberg/pull/24914. (tldr; while it'll introduce some false positives, I still think it's worth enabling it in the long run.)

As for this example, both `setIsListViewOpened` and `isListViewOpenByDefault` should be included. If we don't want to react to changes for `isListViewOpenByDefault`, we could wrap it in a ref to make it explicit.

```js
const isListViewOpenByDefaultRef = useRef( isListViewOpen );

useEffect( () => {
	if ( isListViewOpenByDefaultRef.current ) {
		setIsListViewOpened( true );
	}
}, [ setIsListViewOpened ] );
```

(BTW, shouldn't it make more sense to rename the variable `shouldListViewOpenByDefault`? 😆)

Reply:
On the other hand, I think we could dispatch the action immediately before rendering the React tree to update the state to prevent it from unnecessary re-rendering. IIRC, we do this in the site editor.

Inside `initializeEditor` (or `reinitializeEditor`, not sure), we could dispatch action before starting the render.

```js
function initializeEditor() {
	if ( select( editPostStore ).isFeatureActive( 'showListViewByDefault' ) ) {
		dispatch( editPostStore ).setIsListViewOpened( true );
	}

	render( // ...
}
```

Reply:
Thanks for the detailed notes @kevin940726 🙇 

I just wanted things to run on mount, your `initializeEditor` suggestion sounds promising. I'll look into it.